br/pkg/lightning/mydump/loader.go

```diff
@@ -510,19 +510,25 @@ func (s *mdLoaderSetup) route() error {
 	}
 
 	knownDBNames := make(map[string]*dbInfo)
-	for _, info := range s.dbSchemas {
+	for _, info := range s.tableSchemas {
 		knownDBNames[info.TableName.Schema] = &dbInfo{
 			fileMeta: info.FileMeta,
 			count:    1,
 		}
-	}
-	for _, info := range s.tableSchemas {
 		knownDBNames[info.TableName.Schema].count++
 	}
 	for _, info := range s.viewSchemas {
+		knownDBNames[info.TableName.Schema] = &dbInfo{
```

Comment:
we should check if the item is already in knownDBNames, otherwise we will overwrite existing items.

val, ok := m[key]

Reply:
yes I realized that later too, modified